compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -210,6 +210,18 @@ impl TyKind<'tcx> {
             _ => false,
         }
     }
+
+    /// Get the article ("a" or "an") to use with this type.
+    ///
+    /// **Panics if `self` is [`TyKind::Error`].**
+    pub fn article(&self) -> &'static str {
+        match self {
+            Int(_) | Float(_) | Array(_, _) => "an",
+            Adt(def, _) if def.is_enum() => "an",
+            Error(_) => panic!(),
```

Comment:
I would like to avoid ICEing in this case. It is technically a mistake to show TyErr, but ICEing feels too harsh for it.

Reply:
It would be bad to silently fail on something like that though, right?

Reply:
maybe return an empty string? i think it's a fine compromise in this case

Reply:
I'd rather not silently fail in this case, but I guess it would be bad if this caused someone's code to stop compiling because the compiler crashed. I'm not sure if an empty string makes sense since it will cause a double space in the output. I'll just go with "a" here.